### PR TITLE
Add `other::Expect` header

### PR DESCRIPTION
**Pull request by yoshuawuyts**
Add `other::Expect` header. Ref #99. Thanks!

### CODE REVIEW

**Comment by Fishrock123** [comment]
Is `100-continue` the only relevant value here?

**Comment by yoshuawuyts** [comment]
@Fishrock123 yep, and it seems unlikely any others will be added anytime soon. No known browser currently uses this mechanism; it seems curl is the primary user of this right now.

File: src/other/expect.rs [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+    }
+
+    /// Get the `HeaderName`.
+    pub fn name(&self) -> HeaderName {
+        EXPECT
+    }
+
+    /// Get the `HeaderValue`.
+    pub fn value(&self) -> HeaderValue {
+        let value = "100-continue";
```
Comment (Fishrock123) [inline src/other/expect.rs:71]:
```suggestion
        let value = b"100-continue";
```

**Comment by Fishrock123** [comment]
Should we even implement this? Does Curl require it or something? i.e. is it useful?

**Comment by jbr** [comment]
We use it in async-h1, and when async-h1 client behavior is slightly more mature it should also use this header. I don't really see much value in including it as a reusable type, though. It's just as easy to do `.insert_header("Expect", "100-continue")`

**Comment by yoshuawuyts** [comment]
@jbr I mostly included it for completion. We also have typed headers for e.g. `Content-Length` which is arguably trivial too.

**Comment by Fishrock123** [comment]
Parsing `Content-Length` is not exactly uncommon though.

File: src/other/expect.rs (hunk 1 above)
Comment (yoshuawuyts) [reply src/other/expect.rs:71]: This requires an additional `to_vec` call, which doesn't seem like much of an improvement over calling `into`.

**Comment by yoshuawuyts** [comment]
It seems the conversation now is about useful this header will turn out to be, rather than whether the implementation is correct. I don't see any concrete downsides in including this; the worst case seems to be that nobody will care. With the upside that this might actually be helpful to some.

Making a decision to merge this.

**Pull request merged**